keel-core/src/test/kotlin/com/netflix/spinnaker/keel/actuation/IntermittentFailureTests.kt

```diff
@@ -0,0 +1,170 @@
+    val publisher = mockk<ApplicationEventPublisher>(relaxUnitFun = true)
+    val clock = MutableClock()
+    val vetoRepository = InMemoryUnhappyVetoRepository(clock)
+
+    val dynamicConfigService: DynamicConfigService = mockk(relaxUnitFun = true) {
+      every {
+        // mimicking how a cluster is set up
+        getConfig(Int::class.java, "veto.unhappy.max-diff-count", any())
+      } returns 2
+      every {
+        // mimicking how a cluster is set up
+        getConfig(String::class.java, "veto.unhappy.waiting-time", any())
+      } returns "PT0S" // Duration.ZERO
+    }
+
+    val veto = UnhappyVeto(
+      resourceRepository,
+      diffFingerprintRepository,
+      vetoRepository,
+      dynamicConfigService,
+      "PT10M"
+    )
+    val vetoEnforcer = VetoEnforcer(listOf(veto))
+    val subject = ResourceActuator(
+      resourceRepository,
+      artifactRepository,
+      deliveryConfigRepository,
+      diffFingerprintRepository,
+      listOf(plugin1, plugin2),
+      actuationPauser,
+      vetoEnforcer,
+      publisher,
+      Clock.systemUTC())
+    val desired = DummyResourceSpec(data = "fnord")
+    val current = DummyResourceSpec()
+    val diff = DefaultResourceDiff(desired, current)
+  }
+
+  fun tests() = rootContext<Fixture> {
+
+    fixture { Fixture() }
+
+    before {
+      every { plugin1.name } returns "plugin1"
+      every { plugin1.supportedKind } returns SupportedKind(parseKind("plugin1/foo@v1"), DummyResourceSpec::class.java)
+      every { plugin2.name } returns "plugin2"
+      every { plugin2.supportedKind } returns SupportedKind(parseKind("plugin2/bar@v1"), DummyResourceSpec::class.java)
+    }
+
+    after {
+      resourceRepository.dropAll()
+    }
+
+    context("resource has a diff") {
+      val resource: Resource<DummyResourceSpec> = resource(
+        kind = parseKind("plugin1/foo@v1")
+      )
+      val deliveryConfig = deliveryConfig(resource)
+
+      before {
+        resourceRepository.store(resource)
+        resourceRepository.appendHistory(ResourceCreated(resource))
+        resourceRepository.appendHistory(ResourceValid(resource))
+        coEvery { plugin1.actuationInProgress(resource) } returns false
+        // diff has happened once
+        diffFingerprintRepository.store(resource.id, diff)
+
+        // current state is diff
+        coEvery { plugin1.desired(resource) } returns desired
+        coEvery { plugin1.current(resource) } returns current
+        coEvery { plugin1.update(resource, any()) } returns listOf(Task(id = randomUID().toString(), name = "a task"))
+      }
+      test("diff seen second time, so update happens") {
+        runBlocking { subject.checkResource(resource) }
+        coVerify { plugin1.update(resource, any()) }
+      }
+
+      context("diff seen third time") {
+        before {
+          // diff has happened twice
+          diffFingerprintRepository.store(resource.id, diff)
+        }
+        test("marked unhappy, so no actuation") {
```

Comment:
According to the description here, the test should also check that the status of the resource is unhappy, right?

Reply:
changing the name of the tests so it doesn't have happy in the name